isy-security/src/main/java/de/bund/bva/isyfact/security/oauth2/client/authentication/ClientCredentialsClientRegistrationAuthenticationProvider.java

```diff
@@ -3,60 +3,55 @@
 import org.springframework.lang.Nullable;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.client.ClientAuthorizationException;
 import org.springframework.security.oauth2.client.OAuth2AuthorizationContext;
 import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
 import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProvider;
 import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProviderBuilder;
 import org.springframework.security.oauth2.client.registration.ClientRegistration;
-import org.springframework.security.oauth2.client.registration.ClientRegistrations;
-import org.springframework.security.oauth2.core.AuthorizationGrantType;
-import org.springframework.security.oauth2.core.OAuth2AuthorizationException;
 import org.springframework.security.oauth2.core.OAuth2Error;
 import org.springframework.security.oauth2.core.OAuth2ErrorCodes;
 import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
 
+import de.bund.bva.isyfact.security.oauth2.client.authentication.token.ClientCredentialsClientRegistrationAuthenticationToken;
+
 /**
  * Authentication Provider to obtain an {@link Authentication} with the OAuth2 Client Credentials flow
- * using ad hoc created ClientRegistrations.
+ * using an externally created Client Registration object.
  */
-public class ManualClientCredentialsAuthenticationProvider extends IsyOAuth2AuthenticationProvider {
+public class ClientCredentialsClientRegistrationAuthenticationProvider extends IsyOAuth2AuthenticationProvider {
 
     /**
      * AuthorizedClientProvider for the client credentials.
-     * Unlike in {@link ClientCredentialsAuthenticationProvider} we can't use the AuthorizedClientManager because
+     * Unlike in {@link ClientCredentialsAuthorizedClientAuthenticationProvider} we can't use the AuthorizedClientManager because
      * the client registrations are created manually.
```

Comment:
Hmm, that will not hit the point exactly.
i think it will be more expressive to say:
`because the client registrations are not managed by Spring-configuration properties, but given by parameters`

Reply:
sentence will adapted:
`created programmatically from given runtime-parameters